internal/build/extend_build_test.go

```diff
@@ -0,0 +1,138 @@
+package build_test
+
+import (
+	"bytes"
+	"context"
+	"io"
+
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/apex/log"
+	"github.com/buildpacks/pack/internal/build"
+	"github.com/buildpacks/pack/internal/build/fakes"
+
+	"github.com/buildpacks/pack/pkg/archive"
+	"github.com/buildpacks/pack/pkg/logging"
+	h "github.com/buildpacks/pack/testhelpers"
+	"github.com/docker/docker/api/types"
+	"github.com/golang/mock/gomock"
+	"github.com/heroku/color"
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+
+	mockdocker "github.com/buildpacks/pack/internal/build/mockdocker"
+)
+
+func TestBuildDockerfiles(t *testing.T) {
+	color.Disable(true)
+	defer color.Disable(false)
+	spec.Run(t, "buildExtendByDocker", testBuildDockerfiles, spec.Report(report.Terminal{}), spec.Sequential())
+
+}
+
+func testBuildDockerfiles(t *testing.T, when spec.G, it spec.S) {
+	var (
+		mockDockerClient *mockdocker.MockDockerClient
+		mockController   *gomock.Controller
+		lifecycle        *build.LifecycleExecution
+		tmpDir           string
+
+		// lifecycle options
+		providedClearCache   bool
+		providedPublish      bool
+		providedBuilderImage = "some-registry.com/some-namespace/some-builder-name"
+
+		configureDefaultTestLifecycle func(opts *build.LifecycleOptions)
+		lifecycleOps                  []func(*build.LifecycleOptions)
+	)
+
+	it.Before(func() {
+		var err error
+		mockController = gomock.NewController(t)
+		mockDockerClient = mockdocker.NewMockDockerClient(mockController)
+		h.AssertNil(t, err)
+
+		configureDefaultTestLifecycle = func(opts *build.LifecycleOptions) {
+			opts.BuilderImage = providedBuilderImage
+			opts.ClearCache = providedClearCache
+			opts.Publish = providedPublish
+		}
+
+		lifecycleOps = []func(*build.LifecycleOptions){configureDefaultTestLifecycle}
+
+	})
+
+	when("Extend Build Image By Docker", func() {
+		it("should extend build image using 1 extension", func() {
```

Comment:
Looks good :) let's also test 2 extensions if possible. Ideally we'd want to show that the state from the first extension is persisted to the second extension. This might be a bit tough with our mocks, but we could at least show that the image reference is threaded through.

Reply:
done in the latest commit